### PR TITLE
Fix docker IT failures for 0.7.0-dev.4 vendoring

### PR DESCRIPTION
**Pull request by aboch**
- Fix npe in sbJoin error path
- Fail again endpoint Join in case of failure
  in programming the external connectivity
- In bridge, look for parent and child container configs
  in the generic data
- iptables.Exists() might be called before any other call to
  iptables.raw(). We need to call checkInit() then.

Signed-off-by: Alessandro Boch <aboch@docker.com>

### CODE REVIEW

File: endpoint.go [hunk 1]
```diff
@@ -456,15 +456,23 @@ func (ep *endpoint) sbJoin(sb *sandbox, options ...EndpointOption) error {
 		if extEp != nil {
 			log.Debugf("Revoking external connectivity on endpoint %s (%s)", extEp.Name(), extEp.ID())
 			if err := d.RevokeExternalConnectivity(extEp.network.ID(), extEp.ID()); err != nil {
-				log.Warnf("driver failed revoking external connectivity on endpoint %s (%s): %v",
+				return types.InternalErrorf(
+					"driver failed revoking external connectivity on endpoint %s (%s): %v",
 					extEp.Name(), extEp.ID(), err)
 			}
+			defer func() {
+				if err := d.ProgramExternalConnectivity(extEp.network.ID(), extEp.ID(), sb.Labels()); err != nil {
```
Comment (mavenugo) [inline endpoint.go]: What is the need to unconditionally call `ProgramExternalConnectivity` in the defer ?
That seems incorrect both to the fact that we are doing it unconditionally and also to the fact that the error check cannot be performed with the way `err` variable is managed later and `return sb.clearDefaultGW()` is called without updating the `err` variable.

Comment (aboch) [inline endpoint.go]: The unconditional call is a bug. Thanks for catching it.
The clear default gw has always been a best effort, we have not rolled back a join if that fails.
IMO it should still behave that way,.

**Comment by mavenugo** [comment]
thanks @aboch . LGTM

**Comment by sanimej** [comment]
LGTM

**Pull request merged**